modules/hotmaps/1.0/hotmaps.smk

```diff
@@ -0,0 +1,837 @@
+        mysql_host = CFG["options"]["mysql"]["mysql_host"],
+        mysql_user = CFG["options"]["mysql"]["mysql_user"],
+        mysql_pass = CFG["options"]["mysql"]["mysql_passwd"],
+        mysql_db = CFG["options"]["mysql"]["mysql_db"],
+        mut_regex = "'^input.+\.maf'$",
+        script = CFG["dirs"]["inputs"] + "HotMAPS-master/scripts/mupit/map_maf_to_structure.py"
+    conda:
+        CFG["conda_envs"]["hotmaps"]
+    log:
+        stdout = CFG["logs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/prep_mutations/prep_mutations.stdout.log",
+        stderr = CFG["logs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/prep_mutations/prep_mutations.stderr.log"
+    shell:
+        op.as_one_line("""
+        python {params.script} 
+        --data-dir {params.mut_dir} 
+        --match-regex {params.mut_regex} 
+        --host {params.mysql_host} 
+        --db {params.mysql_db} 
+        --mysql-user {params.mysql_user} 
+        --mysql-passwd {params.mysql_pass} 
+        --output-dir {params.mut_dir} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+rule _hotmaps_prep_mupit_annotation:
+    input:
+        maf = str(rules._hotmaps_input.output.maf)
+    output:
+        annotation = CFG["dirs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/mupit_annotations/mupit_mutations_{sample_set}"
+    params:
+        mut_dir = CFG["dirs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/mutations/",
+        tumor_type = "{sample_set}",
+        mysql_db = CFG["options"]["mysql"]["mysql_db"],
+        mysql_host = CFG["options"]["mysql"]["mysql_host"],
+        mysql_user = CFG["options"]["mysql"]["mysql_user"],
+        mysql_pass = CFG["options"]["mysql"]["mysql_passwd"],
+        cov_dir = CFG["dirs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/",
+        hypermut = "-mt " + CFG["options"]["hotmaps"]["hypermut_threshold"] if CFG["options"]["hotmaps"]["hypermut_threshold"] is not None else "",
+        script = CFG["dirs"]["inputs"] + "HotMAPS-master/scripts/maf/convert_maf_to_mupit.py"
+    conda:
+        CFG["conda_envs"]["hotmaps"]
+    log:
+        stdout = CFG["logs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/prep_mupit_annotation/prep_mupit_annotation.stdout.log",
+        stderr = CFG["logs"]["hotmaps"] + "{sample_set}--{launch_date}/{md5sum}/prep_mupit_annotation/prep_mupit_annotation.stderr.log"
+    shell:
+        op.as_one_line("""
+        python {params.script} 
+        --maf {input.maf} 
+        -mh {params.mysql_host} 
+        -mdb {params.mysql_db} 
+        --mysql-user {params.mysql_user} 
+        --mysql-passwd {params.mysql_pass} 
+        --tumor-type {params.tumor_type} 
+        --no-stratify {params.hypermut} 
+        -i {params.cov_dir} 
+        --output {output.annotation} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+rule _hotmaps_filter_hypermutated:
```

Comment:
what's the purpose of this? Can we document it?

Reply:
I added a description above the rule